### ISSUE TITLE
proc macro span error information missing after rustfmt breaks 

### ISSUE THREAD

**Issue by MabezDev** [rust-lang/rust#56894]
Before rustfmt:
```rust
static mut TOUCH: hal::tsc::Tsc<hal::gpio::gpiob::PB4<hal::gpio::Alternate<hal::gpio::AF9, hal::gpio::Output<hal::gpio::OpenDrain>>>> = ();
```
after:
```rust
static mut TOUCH: hal::tsc::Tsc<
        hal::gpio::gpiob::PB4<
            hal::gpio::Alternate<hal::gpio::AF9, hal::gpio::Output<hal::gpio::OpenDrain>>,
        >,
    > = ();
```

These fields are contained in the proc macro - specifically [cortex-m-rtfm](https://github.com/japaric/cortex-m-rtfm) the error information is reported at the start of proc macro attribute.

e.g
```rust
error[E0308]: mismatched types
  --> src/main.rs:64:1
   |
64 | #[app(device = crate::stm32l4x2)]
   | ^ expected (), found u32
   |
   = note: expected type `()`
              found type `u32`
```

even though error is produced much further along in the macro

**Comment by MabezDev** [comment]
I had a rust-toolchain pinning my rustc to an old nightly (months back). Removing this and using a recent nightly has fixed my issue.
